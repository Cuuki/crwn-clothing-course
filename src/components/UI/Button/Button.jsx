import React from 'react';

import {ButtonStyled} from './Button.styles';

const Button = ({children, ...props}) => (
  <ButtonStyled className="custom-button" {...props}>
    {children}
  </ButtonStyled>
);

export default Button;
