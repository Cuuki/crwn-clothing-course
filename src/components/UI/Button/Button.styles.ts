import styled, {css} from 'styled-components';

const defaultButtonStyles = css`
  border: none;
  background-color: black;
  color: white;

  &:hover {
    border: 1px solid black;
    background-color: white;
    color: black;
  }
`;

const invertedButtonStyles = css`
  border: 1px solid black;
  background-color: white;
  color: black;

  &:hover {
    background-color: black;
    color: white;
    border: none;
  }
`;

const googleButtonStyles = css`
  background-color: #4285f4;
  color: white;

  &:hover {
    background-color: #357ae8;
    border: none;
  }
`;

const getButtonStyles = props => {
  if (props.isGoogleSignIn) {
    return googleButtonStyles;
  }

  return props.inverted ? invertedButtonStyles : defaultButtonStyles;
};

export const ButtonStyled = styled.button`
  min-width: 165px;
  width: auto;
  height: 50px;
  letter-spacing: 0.5px;
  line-height: 50px;
  padding: 0 35px 0 35px;
  font-size: 15px;
  text-transform: uppercase;
  font-family: 'Open Sans Condensed';
  font-weight: bolder;
  cursor: pointer;
  text-align: center;

  ${getButtonStyles}
`;
