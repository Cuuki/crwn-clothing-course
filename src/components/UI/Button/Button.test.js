import React from 'react';
import {render} from "@testing-library/react";
import Button from './Button';

it('renders Button component', () => {
  expect(
    render(
      <Button onClick={() => {
      }} inverted isGoogleSignIn>
        Test
      </Button>
    )
  ).toMatchSnapshot();
});
