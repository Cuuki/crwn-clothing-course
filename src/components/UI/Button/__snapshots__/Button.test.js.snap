// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component 1`] = `
<styled.button
  className="custom-button"
  inverted={true}
  isGoogleSignIn={true}
  onClick={[Function]}
>
  Test
</styled.button>
`;
